hathor/indexes/rocksdb_mempool_tips_index.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Iterable, Iterator, Optional, Set, cast
+
+from structlog import get_logger
+
+from hathor.indexes.mempool_tips_index import MempoolTipsIndex
+from hathor.indexes.rocksdb_utils import RocksDBSimpleSet
+from hathor.transaction import BaseTransaction, Transaction
+from hathor.transaction.storage.traversal import BFSWalk
+from hathor.util import not_none
+
+if TYPE_CHECKING:  # pragma: no cover
+    import rocksdb
+
+    from hathor.transaction.storage import TransactionStorage
+
+logger = get_logger()
+
+_CF_NAME_MEMPOOL_TIPS_INDEX = b'mempool-tips-index'
+
+
+class RocksDBMempoolTipsIndex(MempoolTipsIndex):
+    _index: RocksDBSimpleSet
+
+    def __init__(self, db: 'rocksdb.DB', *, cf_name: Optional[bytes] = None) -> None:
+        self.log = logger.new()
+        _cf_name = cf_name or _CF_NAME_MEMPOOL_TIPS_INDEX
+        self._index = RocksDBSimpleSet(db, self.log, cf_name=_cf_name)
+
+    def _remove_many(self, remove_txs: Iterable[bytes]) -> None:
+        for tx in iter(remove_txs):
+            if tx in self._index:
+                self._index.remove(tx)
+
+    def update(self, tx: BaseTransaction) -> None:
```

Comment:
That makes sense. But I'd like to avoid abstracting "internal" methods. I the base module (`hathor.indexes.mempool_tips_index`) could provide a helper method/function with all of this logic that the concrete classes can choose to use to implement this method. I'll make a new commit with what I have in mind.

Reply:
Done.